### DISCUSSION TITLE
Problem with filter results - Need to refresh

### DISCUSSION THREAD

**Discussion by lironsam** [rappasoft/laravel-livewire-tables#374] [Q&A]
Hi,

I have implemented a filter that match a specific label.

`->when($this->getFilter('color'), fn ($query, $color) => $query->where('color', $color))`

It works the first time. Then, if I select another color, the results are messed up i.e I have not only this color but others colors labels that are shown. But If I refresh the page, the filter works again.

Also, when there are no results, I have actually the message "No items found. Try narrowing your search.", but I still have some results displayed.

Any ideas why it happens ?

Thank you

**Comment by lrljoe** [comment]
Suggest using the filter() callback on your Filter to prevent this issue.
